packages/flutter_tools/lib/src/base/error_handling_io.dart

```diff
@@ -762,6 +766,19 @@ void _handlePosixException(Exception e, String? message, int errorCode, String?
   _throwFileSystemException(errorMessage);
 }
 
+void _handleMacOSException(Exception e, String? message, int errorCode, String? posixPermissionSuggestion) {
+  // https://github.com/apple/darwin-xnu/blob/master/bsd/dev/dtrace/scripts/errno.d
+  const int ebadarch = 86;
+  if (errorCode == ebadarch) {
+    final StringBuffer errorBuffer = StringBuffer();
+    errorBuffer.writeln(message);
+    errorBuffer.writeln('Flutter requires the Rosetta translation environment on ARM Macs. Try running:');
```

Comment:
This can also happen on x64 when an ARM executable is installed https://github.com/flutter/flutter/issues/115691.
Unfortunately we don't have access to `operatingSystemUtils.hostPlatform` here to switch on that.

Reply:
I changed the wording since we don't know if they are on an ARM Mac:
```
This binary was built with the incorrect architecture to run on this machine.
Flutter requires the Rosetta translation environment. If you are on an ARM Mac, try running:
  sudo softwareupdate --install-rosetta --agree-to-license
```